src/internal/FluencePeer.ts

```diff
@@ -0,0 +1,350 @@
+     * Will be generated randomly if not specified
+     */
+    peerId?: string;
+
+    /**
+     * When the peer established the connection to the network it sends a ping-like message to check if it works correctly.
+     * The options allows to specify the timeout for that message in milliseconds.
+     * If not specified the default timeout will be used
+     */
+    checkConnectionTimoutMs?: number;
+
+    /**
+     * When the peer established the connection to the network it sends a ping-like message to check if it works correctly.
+     * If set to true, the ping-like message will be skipped
+     * Default: false
+     */
+    skipCheckConnection?: boolean;
+
+    /**
+     * The dialing timeout in milliseconds
+     */
+    dialTimeoutMs?: number;
+}
+
+/**
+ * Information about Fluence Peer connection
+ */
+interface ConnectionInfo {
+    /**
+     * Is the peer connected to network or not
+     */
+    isConnected: Boolean;
+
+    /**
+     * The Peer's identification in the Fluence network
+     */
+    selfPeerId: PeerIdB58;
+
+    /**
+     * The list of relays's peer ids to which the peer is connected to
+     */
+    connectedRelays: Array<PeerIdB58>;
+}
+
+/**
+ * This class implements the Fluence protocol for javascript-based environments.
+ * It provides all the necessary features to communicate with Fluence network
+ */
+export class FluencePeer {
+    // TODO:: implement api alongside with multi-relay implementation
+    //async addConnection(relays: Array<ConnectionSpec>): Promise<void> {}
+
+    // TODO:: implement api alongside with multi-relay implementation
+    //async removeConnections(relays: Array<ConnectionSpec>): Promise<void> {}
+
+    /**
+     * Creates a new Fluence Peer instance. Does not start the workflows.
+     * In order to work with the Peer it has to be initialized with the `init` method
+     */
+    constructor() {}
+
+    /**
+     * Get the information about Fluence Peer connections
+     */
+    get connectionInfo(): ConnectionInfo {
+        const isConnected = this._connection?.isConnected();
+        return {
+            isConnected: isConnected,
+            selfPeerId: this._selfPeerId,
+            connectedRelays: this._relayPeerId ? [this._relayPeerId] : [],
+        };
+    }
+
+    /**
+     * Initializes the peer: starts the Aqua VM, initializes the default call service handlers
+     * and (optionally) connect to the Fluence network
+     * @param config - object specifying peer configuration
+     */
+    async init(config?: PeerConfig): Promise<void> {
+        let peerId;
+        const peerIdOrSeed = config?.peerId;
+        if (!peerIdOrSeed) {
+            peerId = await randomPeerId();
+        } else if (isPeerId(peerIdOrSeed)) {
+            // keep unchanged
+            peerId = peerIdOrSeed;
+        } else {
+            // peerId is string, therefore seed
+            peerId = await peerIdFromEd25519SK(peerIdOrSeed);
```

Comment:
What will happen if `peerIdOrSeed` is malformed?

Reply:
We decided to replace this logic with the help of KeyPair class